pennylane/operation.py

```diff
@@ -2037,6 +2039,12 @@ def data(self):
         """
         return sum((o.data for o in self.obs), [])
 
+    @data.setter
+    def data(self, new_data):
```

Comment:
Some documentation about what the shape of the new data should look like might be helpful. This is bound to be a developer-facing setter, but devs could assume that the new data list should only include data for observables that have parameters rather than new data for parametrized observables as well as empty lists for non-parametrized observables. For example, without documentation, one could use this setter in the following way and expect it to behave correctly:

```pycon
>>> t = qml.Tensor(qml.PauliX(0), qml.Hermitian(np.eye(2), wires=1))
>>> t.data = [np.array([[1, 1], [1, -1]])]
```

Meanwhile, the expected behaviour is:
```pycon
>>> t = qml.Tensor(qml.PauliX(0), qml.Hermitian(np.eye(2), wires=1))
>>> t.data = [[], np.array([[1, 1], [1, -1]])]
```

Reply:
Added docstring with an example!